docs/ai-instructions/migrate-all-cartridges-instructions.md

```diff
@@ -0,0 +1,148 @@
+
+Additionally some other renamings have to be done in `*.component` files:
+| Old | New |
+|-----|-----|
+|intershop.B2BWebShop.RESTAPI.basket.v1.BasketQuoteListResource|intershop.WebShop.RESTAPI.BasketQuoteListResource|
+|intershop.B2BWebShop.RESTAPI.b2c.CustomerListResource|intershop.WebShop.RESTAPI.PrivateCustomerListResource|
+|com.intershop.sellside.rest.b2c.capi.resource.customer.PrivateCustomerItemResource|com.intershop.sellside.rest.b2c.capi.resource.customer.PrivateCustomerItemResource|
+
+Also check all `*.java` files if the following pattern is used `@Named("intershop.WebShop.RESTAPI.*")`. This has to be adapted accordingly.
+
+### 10. Change Registration of cartridges to application-specific cartridge lists
+In case a cartridge contains files `app-extension.component` or `apps-extension.component` the content of this file must be splitted up and moved to different `apps.component` files in different `as_` cartridges.
+The location for the `apps.component` is `src/main/resources/resources/<cartridge>/components`.
+
+All lines of `app-extension.component` or `apps-extension.component` containing `<fulfill requirement="selectedCartridge"` have to be moved. The used cartridge-list (see table) determines to which cartridge the line has to be moved.
+Before a line can be moved it has to be checked if there's is already a cartridge with the cartridge prefix from table. If the cartridge does not exist it must be created.
+Its name is the prefix from the table `cartridge prefix` concatenated with a project name which the user of this script should enter. 
+When a line is moved, the cartridge must be added as runtime dependency in the `build.gradle.kts` of the `as_` cartridge.
+
+| cartridge-list | cartridge prefix|
+|----------------|-----------------|
+|intershop.REST.Cartridges|as_headless_|
+|intershop.EnterpriseBackoffice.Cartridges|as_backoffice_|
+|intershop.SLDSystem.Cartridges|as_sldsystem_|
+|intershop.SMC.Cartridges|as_smc_|
+
+### 11. Dependency Configuration Clarification
+
+| Need | Use | Notes |
+|------|-----|-------|
+| Compile & runtime access to another cartridge | `cartridge(project(":other"))` | Replaces old `compile project()` |
+| Runtime-only presence of another cartridge (assembly style) | `cartridgeRuntime(project(":other"))` | Only if you truly do NOT compile against its code |
+| Intershop provided cartridge (group starts with `com.intershop.`) | `cartridge("com.intershop...:artifact")` or `cartridgeRuntime(...)` | Same rule as above re: compile vs runtime |
+| Third-party library | `implementation("group:artifact:version")` | Keep versions centralized if possible |
+| Annotation processors | `annotationProcessor("com.intershop.platform:...")` | Only when required annotations present |
+
+### 12. Cleanup Checklist
+* Delete obsolete `*.version` files in root directory
```

Comment:
How should the AI know if they are obsolete? I think some special instructions would be helpful (content has to be covered by the `build.gradle.kts` in the `versions` directory)